pkg/config/documentation.go

```diff
@@ -0,0 +1,50 @@
+package config
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/lunarway/shuttle/pkg/errors"
+	"github.com/lunarway/shuttle/pkg/git"
+)
+
+// DocumentationURL returns a URL pointing to plan documentation if any is
+// available. Plan reference and plan documentation field is inspected and
+// parsed.
+func (p *ShuttleProjectContext) DocumentationURL() (string, error) {
+	var ref string
+	switch {
+	case p.Plan.Documentation != "":
+		ref = p.Plan.Documentation
+	case p.Config.Plan != "":
+		ref = p.Config.Plan
+	default:
+		return "", errors.NewExitCode(1, "Could not find any plan documentation")
+	}
+
+	switch {
+	case git.IsPlan(ref):
+		return normalizeGitPlan(git.ParsePlan(ref))
+	case isMatching("^(http|https)://", ref):
```

Comment:
Can we extract the regex to a variable with a meaningful name. That will make it easier for me to quickly understand what is happening.

Reply:
Sure 👍